Jenkinsfile

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+pipeline {
+  agent none
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    timeout(time: 8, unit: 'HOURS')
+  }
+  triggers {
+    cron('@weekly')
+    pollSCM('@daily')
+  }
+  stages {
+    stage ('Debug') {
+      options {
+        timeout(time: 1, unit: 'HOURS')
+        retry(2)
+      }
+      agent {
+        docker {
+          label 'ubuntu'
+          image 'apachedirectory/maven-build:jdk-8'
+          args '-v $HOME/.m2:/home/hnelson/.m2'
+        }
+      }
+      steps {
+        sh 'env'
+      }
+      post {
+        always {
+          deleteDir()
+        }
+      }
+    }
+    stage ('Build and Test') {
+      parallel {
+        stage ('Linux Java 8') {
+          options {
+            timeout(time: 4, unit: 'HOURS')
+            retry(2)
+          }
+          agent {
+            docker {
+              label 'ubuntu'
+              image 'apachedirectory/maven-build:jdk-8'
+              args '-v $HOME/.m2:/home/hnelson/.m2'
```

Comment:
Who is "hnelson"? Should this be in the PR?

Reply:
Yes. That's the username in the custom docker image that is used for building the other `Directory` projects. The username is used in various Kerberos/GSSAPI integration tests. https://github.com/apache/directory-buildtools/tree/master/docker/maven-build

Reply:
ok thanks